internal/mode/static/nginx/config/http/config.go

```diff
@@ -7,6 +7,7 @@ type Server struct {
 	Locations     []Location
 	IsDefaultHTTP bool
 	IsDefaultSSL  bool
+	HTTP2         bool
```

Comment:
I think we should support disallowing HTTP/2.
This way, users can disable it if they don't need it.
Additionally, HTTP/2 implementation can be a source of vulnerabilities -- https://nginx.org/en/security_advisories.html --  so disabling it could be a quick mitigation strategy.

Reply:
Any idea how we should approach this? In my testing, if it wasn't set in the http context (so hardcoded in the nginx.conf), HTTP/1 was being used by default even if we it was enabled in the server context. Does it work to disable it in any server by default unless a gRPC route is configured?

Reply:
I did some further testing too.

Looks like if we don't enable `http2` for all servers on that listener port, 
`grpcurl` will not work, because NGINX will not allow HTTP/2 without upgrading to it, which it doesn't do.

grpcurl testing is an equivalent to `curl foo.bar.com:8080 -v --http2-prior-knowledge` , which NGINX will get `curl: (56) Failure when receiving data from the peer`

So yep, I think we need to enable HTTP/2 everywhere.
We can still enable it per port (ex. in all servers that listen on port 80).

I think enabling HTTP/2 should be a separate settings, with an option to be disabled. In the later case, NGF will not allow GRPCRoutes.

Reply:
So enable by default right, so we support GRPCRoute out of the box? Maybe a command line argument for now? It would be a perfect one for the NGINX Proxy CRD but that will require some design.

@kate-osborn @sjberman any thoughts?

Reply:
If we put it in the NginxProxy CRD, then it applies to all nginx data planes under the GatewayClass. Is this too heavy handed? A CLI argument would do the same, but could be a more "temporary" solution that we can deprecate and phase out in the future when we figure out where it goes in the API.